### PR TITLE
Add Kafka Avro encoder

### PR DESCRIPTION
**Pull request by charlesjmorgan**
Add `AvroRowEncoder` and `AvroRowEncoderFactory`
Add test case in `io.prestosql.plugin.kafka.TestKafkaIntegrationSmokeTest#testRoundTripAllFormats`

### CODE REVIEW

**Comment by losipiuk** [comment]
Make commit message just `Add Kafka avro encoder`

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/RowEncoder.java [hunk 1]
```diff
@@ -15,7 +15,10 @@
 
 import io.prestosql.spi.block.Block;
 
+import java.io.Closeable;
+
 public interface RowEncoder
+        extends Closeable
```
Comment (losipiuk) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/RowEncoder.java:21]: Extract making `RowEncoder` Closable to separate commit.

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+public class AvroRowEncoder
+        extends AbstractRowEncoder
+{
+    private static final Set<Type> SUPPORTED_PRIMITIVE_TYPES = ImmutableSet.of(
+            BOOLEAN, INTEGER, BIGINT, DOUBLE, REAL);
+
+    public static final String NAME = "avro";
+
+    private final ByteArrayOutputStream byteArrayOuts;
```
Comment (losipiuk) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java:51]: rename to `byteArrayOutputStream`

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+    @Override
+    public byte[] toByteArray()
+    {
+        // make sure entire row has been updated with new values
+        checkArgument(currentColumnIndex == columnHandles.size(), format("Missing %d columns", columnHandles.size() - currentColumnIndex + 1));
+
+        try {
+            byteArrayOuts.reset();
+            dataFileWriter.create(parsedSchema, byteArrayOuts);
```
Comment (losipiuk) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java:138]: do we need to call this one for each row?
And if should we ignore `DataFileWriter` it returns?

File: presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationSmokeTest.java [hunk 4]
```diff
@@ -141,6 +141,10 @@ public void testRoundTripAllFormats(RoundTripTestCase testCase)
     private List<RoundTripTestCase> testRoundTripAllFormatsData()
     {
         return ImmutableList.<RoundTripTestCase>builder()
+                .add(new RoundTripTestCase(
```
Comment (losipiuk) [inline presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationSmokeTest.java:144]: can we write more than single row in the test?

(this actually applies to other formats too)

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java (hunk 3 above)
Comment (aalbu) [reply presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java:138]: This is not a pretty API.  It looks like this call gets the writer ready to write to a new output stream with a new schema.  Feels like a factory method, but it just resets the instance.

And without the schema registry integration, writing Avro messages are quite heavyweight (each message contains the schema).

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+            throw new UncheckedIOException("Failed to append record", e);
+        }
+    }
+
+    @Override
+    public void close()
+    {
+        try {
+            byteArrayOuts.close();
+            dataFileWriter.close();
```
Comment (aalbu) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java:155]: Nothing wrong with this, but it didn't seem necessary.  `ByteArrayOutputStream.close()` is really a no-op and I don't think the Avro writer holds on to any resources.

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java (hunk 3 above)
Comment (charlesjmorgan) [reply presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoder.java:138]: It doesn't work without the `create` method call

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/RowEncoder.java (hunk 1 above)
Comment (losipiuk) [reply presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/RowEncoder.java:21]: Sorry for not being precise.
I meant extracting the interface chage to separate commit which is before one which add `Avro encoder`.

Then `AvroRowEncoder.close()` can be added together with rest of Avro encoder code.

File: presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoderFactory.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class AvroRowEncoderFactory
+        implements RowEncoderFactory
+{
+    @Override
+    public RowEncoder create(ConnectorSession session, Optional<String> dataSchema, List<EncoderColumnHandle> columnHandles)
+    {
+        checkArgument(dataSchema.isPresent(), "dataSchema for avro format is not present");
```
Comment (losipiuk) [inline presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/avro/AvroRowEncoderFactory.java:34]: s/avro/Avro/

**Pull request merged**